clients/shared/append.go

```diff
@@ -1,41 +1,29 @@
-package snowflake
+package shared
 
 import (
+	"fmt"
 	"log/slog"
 
+	"github.com/artie-labs/transfer/lib/config"
 	"github.com/artie-labs/transfer/lib/config/constants"
+	"github.com/artie-labs/transfer/lib/destination"
 	"github.com/artie-labs/transfer/lib/destination/ddl"
 	"github.com/artie-labs/transfer/lib/destination/types"
-	"github.com/artie-labs/transfer/lib/typing/columns"
-
 	"github.com/artie-labs/transfer/lib/optimization"
+	"github.com/artie-labs/transfer/lib/typing/columns"
 )
 
-func (s *Store) Append(tableData *optimization.TableData) error {
-	var err error
-	for i := 0; i < maxRetries; i++ {
-		if i > 0 {
-			if IsAuthExpiredError(err) {
-				slog.Warn("Authentication has expired, will reload the Snowflake store and retry appending", slog.Any("err", err))
-				s.reestablishConnection()
-			} else {
-				break
-			}
-		}
-
-		err = s.append(tableData)
-	}
-
-	return err
-}
-
-func (s *Store) append(tableData *optimization.TableData) error {
+func Append(dwh destination.DataWarehouse, tableData *optimization.TableData, cfg config.Config, opts types.AppendOpts) error {
 	if tableData.ShouldSkipUpdate() {
 		return nil
 	}
 
-	fqName := s.ToFullyQualifiedName(tableData, true)
-	tableConfig, err := s.GetTableConfig(tableData)
+	if err := opts.Validate(); err != nil {
```

Comment:
I feel like validating the options should happen before the `tableData.ShouldSkipUpdate()` because presumably if the options are invalid it will cause runtime issues so we want to catch it as soon as possible.

Reply:
Fine by me